xml/xml_utils.go

```diff
@@ -0,0 +1,39 @@
+package xml
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// GetResponseErrorCode returns the error code from an xml error response body
+func GetResponseErrorCode(r io.Reader, noErrorWrapping bool) (string, error) {
+	rb, err := ioutil.ReadAll(r)
+	if err != nil {
+		return "", err
+	}
+
+	if noErrorWrapping {
+		var errResponse errorWrapper
+		err := xml.Unmarshal(rb, &errResponse)
+		if err != nil {
+			return "", fmt.Errorf("error while fetching xml error response code: %w", err)
+		}
+		return errResponse.Code, err
+	}
+
+	var errResponse errorResponse
+	if err := xml.Unmarshal(rb, &errResponse); err != nil {
+		return "", fmt.Errorf("error while fetching xml error response code: %w", err)
+	}
+	return errResponse.Err.Code, nil
```

Comment:
These cases seem backward. Shouldn't the `errorWrapper` be the type used when `noErrorWrapping` is false? it looks like `errorWrapper` is used when there should not be an error wrapper.  Maybe this is more a naming issue. 

because below i'd expect `errorResponse` to of been named `errorWrapper`, and `errorWrapper` to be `errorResponse`.